docs/changelog.rst

```diff
@@ -6,6 +6,10 @@ Changelog
 Development
 ===========
 - (Fill this out as you fix issues and develop your features).
+- BREAKING CHANGE: "class_check" and "read_preference" keyword arguments are no longer available when filtering a QuerySet #2112
+  - Instead of `Doc.objects(foo=bar, read_preference=...)` use `Doc.objects(foo=bar).read_preference(...)`.
+  - Instead of `Doc.objects(foo=bar, class_check=False)` use `Doc.objects(foo=bar).clear_cls_query(...)`.
```

Comment:
The problem with these 2 breaking changes is that they'll silently fail and could lead to bad bugs. Maybe we collect those 2 param in __call__ and raise an error explicitly (and mention the alternative). If you go this way, add a comment to mention that the check is meant to be removed in > 0.20 for instance

Reply:
I'm mostly concerned about `read_preference`, I doubt many people use class_check

Reply:
Wouldn't it throw an `InvalidQueryError` since it's an invalid field?

Reply:
my bad, you are absolutely right

Reply:
Yep, `InvalidQueryError` would be raised in such cases. That being said, since it's a runtime exception, I'm not opposed to releasing one version of ME that deprecates it (throwing a `DeprecationWarning`) and then another release that removes it.

Shall we do that @thomasst @bagerard?

Reply:
I'm okay with just removing it. You'd hopefully have tests catch it.